Common++/header/IpAddresses.h

```diff
@@ -0,0 +1,551 @@
+		 */
+		bool operator!=(const IPAddress& rhs) const { return !(*this == rhs); }
+
+
+		// Following methods are placed for backward compatibility and will be deleted in the future
+
+		/**
+		 * Clone the object
+		 * Notice this method is deprecated and will be deleted in the future
+		 * @return A newly allocated instance which is a clone of the current instance
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is now unnecessary and added for backward compatibility. It will be deleted in the future")]]
+		#endif
+		IPAddress *clone() const { return new IPAddress(*this); }
+
+		/**
+		 * Compare between this IP address and another IP address
+		 * Notice this method is deprecated and will be deleted in the future. Prefer an equal-to operator
+		 * @return True if addresses match or false otherwise
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is now unnecessary and added for backward compatibility. It will be deleted in the future. Prefer an equal-to operator")]]
+		#endif
+		bool equals(const IPAddress* other) const { return *this == *other; }
+
+	private:
+		uint8_t m_Type;
+		IPv4Address m_IPv4;
+		IPv6Address m_IPv6;
+	};
+
+
+	// implementation of inline methods
+
+	bool IPAddress::operator==(const IPAddress& rhs) const
+	{
+		if (isIPv4())
+			return rhs.isIPv4() ? (m_IPv4 == rhs.m_IPv4) : false;
+
+		return m_IPv6 == rhs.m_IPv6;
+	}
+
+	IPAddress& IPAddress::operator=(const IPv4Address& addr)
+	{
+		m_Type = IPv4AddressType;
+		m_IPv4 = addr;
+		return *this;
+	}
+
+	IPAddress& IPAddress::operator=(const IPv6Address& addr)
+	{
+		m_Type = IPv6AddressType;
+		m_IPv6 = addr;
+		return *this;
+	}
+
+
+	// Creation of IPAddress
+
+	/**
+	 * A function that creates an instance of the class out of string (char*) value
+	 * If the string doesn't represent a valid IPv6 or IPv4 address an errorCode will be set to non-zero value
+	 * @param[in] addressAsString The string (char*) representation of the address
+	 * @param[out] errorCode Contains 0 if a string represents a valid address, otherwise a non-zero value
+	 * @return An instance of class IPAddress. If an error occured the return value will contain an unspecified IPv4Address address
+	 */
+	IPAddress makeAddress(const char* addrAsString, int& errorCode);
+
+	/**
+	 * A function that creates an instance of the class out of std::string value
+	 * If the string doesn't represent a valid IPv6 or IPv4 address an errorCode will be set to non-zero value
+	 * @param[in] addressAsString The std::string representation of the address
+	 * @param[out] errorCode Contains 0 if a string represents a valid address, otherwise a non-zero value
+	 * @return An instance of class IPAddress. If an error occured the return value will contain an unspecified IPv4Address address
+	 */
+	inline IPAddress makeAddress(const std::string& addrAsString, int& errorCode) { return makeAddress(addrAsString.c_str(), errorCode); }
```

Comment:
see similar comments for `IPv4Address` and `IPv6Address`

Reply:
Done